server/pulp/server/webservices/views/dispatch.py

```diff
@@ -0,0 +1,102 @@
+from datetime import datetime
+
+from django.views.generic import View
+from mongoengine.queryset import DoesNotExist
+
+from pulp.server.async import tasks
+from pulp.server.auth import authorization
+from pulp.server.db.model.dispatch import TaskStatus
+from pulp.server.db.model.resources import Worker
+from pulp.server.exceptions import MissingResource
+from pulp.server.webservices import serialization
+from pulp.server.webservices.controllers.decorators import auth_required
+from pulp.server.webservices.views.util import (generate_json_response,
+                                                generate_json_response_with_pulp_encoder)
+
+
+def task_serializer(task):
+    """
+    Update the task representation in the database to match the model for the API
+
+    :param task: The task from the database
+    :type task: dict
+    :return: the same task modified for use by the API
+    :rtype: dict
+    """
+    task = serialization.dispatch.task_status(task)
+    task.update(serialization.dispatch.spawned_tasks(task))
+    task.update(serialization.dispatch.task_result_href(task))
+    return task
+
+
+class TaskCollectionView(View):
+    """
+    View for all tasks.
+    """
+
+    @auth_required(authorization.READ)
+    def get(self, request):
+        """
+        Return a response containing a list of all tasks or a response containg a list of all tasks
+        that match optional tags passed in as GET parameters.
+
+        :param request: WSGI request object
```

Comment:
On 02/11/2015 11:37 AM, bmbouter wrote:
> vertically aligned colons! :cry:

I personally like the vertically aligned colons because they
significantly increase readability. I don't like code to be vertically
aligned because later commits might touch more lines than what they need
to, and since code is harder to read than English text I think it's
important to keep those changes smaller. However, I think the risk of
later commits changing comment alignment is worth the readability.